daemon/src/ipc_server.rs

```diff
@@ -146,6 +146,24 @@ pub fn handle_message(
             }
             IpcResponse::Ok
         }),
+
+        IpcMessage::TogglePauseWallpaper { monitors } => {
+            check_monitors(wpaperd, &monitors).map(|_| {
+                let surfaces = collect_surfaces(wpaperd, monitors);
+                if surfaces.is_empty() {
+                    return IpcResponse::Ok;
+                }
+
+                // Ensure synced pause state for the provided surfaces
+                if surfaces[0].should_pause() {
```

Comment:
Just one nitpick, can we hide this inside a `toggle` method in `Surface`? It seems a bit clearer to me. What do you think?

Reply:
Ah, I just noticed. This only checks for the first surface and then apply the same behavior to `surfaces`, while it should toggle the timer depending on each surface specific state.

Reply:
I did think about toggling it per surface, but is that how toggling should work? The idea behind this approach was to keep all the surfaces in sync. Happy to change it to just a method that gets called per surface though

Reply:
That's a good point, keeping the surfaces in sync is also a good approach. However, I think that `pause`, `resume` and `toggle` commands behavior should be consistent between each other.  My guess is that most users would pause/resume using toggle command and all monitors, the additional arguments are for very specific use cases. Can you please call the `toggle()` method per surface?